src/common/cockpitmemfdread.c

```diff
@@ -137,3 +139,27 @@ cockpit_memfd_read_from_envvar (gchar      **result,
   *result = content;
   return TRUE;
 }
+
+JsonObject *
+cockpit_memfd_read_json (gint fd,
+                         GError **error)
+{
+  g_autofree gchar *content = cockpit_memfd_read (fd, error);
+
+  if (content == NULL)
+    return NULL;
+
+  return cockpit_json_parse_object (content, -1, error);
+}
+
+JsonObject *
+cockpit_memfd_read_json_from_control_messages (CockpitControlMessages  *ccm,
```

Comment:
Please rebase your branch so that the commit that introduces `CockpitControlMessages` comes before this one ("memfdread: add a couple more public accessors"). Right now it's both confusing and non-bisectable.